merge/conflict.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package merge
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/structured-merge-diff/fieldpath"
+)
+
+// Conflict is a conflict on a specific field with the current owner of
```

Comment:
Won't you want a list of conflicts as the error type?

Reply:
We could, we don't have too. I suspect the way we want to print the conflicts might lead that decision (though it doesn't have to, we can aggregate/split at any time). It might look like this:
```
conflict with 'random-id': .spec.replicas
conflict with 'random-id': .spec.strategy.type
```

or it could look like that:
```
conflict with 'random-id': 
- .spec.replicas
- .spec.strategy.type
```

Both look fine to me (I like the former better though). 

From a code standpoint, I'd keep each conflict as an individual entity.

Eventually, we'll want to create a list of conflicts from the intersection of owner fieldsets, which means we'll need a: `ConflictsFromFieldset(fieldset) []Conflict`.